src/eager/mod.rs

```diff
@@ -0,0 +1,835 @@
+                x.0.as_ref()
+                    .map(|dims| dims.iter().map(|x| (*x).unwrap_or(-1)).collect())
+            })
+            .collect();
+        let mut ptrs: Vec<*const i64> = c_dims
+            .iter()
+            .map(|x| match *x {
+                None => ptr::null(),
+                Some(ref dims) => dims.as_ptr(),
+            })
+            .collect();
+        let lens: Vec<c_int> = value
+            .iter()
+            .map(|x| match x.0 {
+                None => -1,
+                Some(ref dims) => dims.len() as c_int,
+            })
+            .collect();
+        unsafe {
+            tf::TFE_OpSetAttrShapeList(
+                self.inner,
+                c_attr_name.as_ptr(),
+                ptrs.as_mut_ptr(),
+                lens.as_ptr(),
+                ptrs.len() as c_int,
+                status.inner,
+            );
+        }
+
+        if status.is_ok() {
+            return Ok(());
+        }
+        Err(status)
+    }
+
+    /// Sets a tensor-valued attribute.
+    fn set_attr_any_tensor(&mut self, attr_name: &str, value: &dyn AnyTensor) -> Result<()> {
+        let c_attr_name = CString::new(attr_name)?;
+        let mut status = Status::new();
+        unsafe {
+            tf::TFE_OpSetAttrTensor(
+                self.inner,
+                c_attr_name.as_ptr(),
+                value.inner()?,
+                status.inner(),
+            );
+        }
+        status.into_result()
+    }
+}
+
+impl Drop for Op {
+    fn drop(&mut self) {
+        unsafe { tf::TFE_DeleteOp(self.inner) };
+    }
+}
+
+///
+pub trait ToHandle<'a> {
+    ///
+    fn to_handle(&self, ctx: &'a Context) -> Result<TensorHandle<'a>>;
+}
+
+impl<'a, T> ToHandle<'a> for Tensor<T>
+where
+    T: TensorType,
+{
+    fn to_handle(&self, ctx: &'a Context) -> Result<TensorHandle<'a>> {
+        TensorHandle::new(ctx, self)
+    }
+}
+
+impl<'a> ToHandle<'a> for TensorHandle<'a> {
+    fn to_handle(&self, _: &'a Context) -> Result<TensorHandle<'a>> {
+        self.copy_sharing_tensor()
+    }
+}
+#[cfg(test)]
+mod test {
+
+    use super::*;
+    use crate::Tensor;
+
+    #[test]
+    fn create() {
+        let opts = ContextOptions::new();
+        let ctx = Context::new(opts).unwrap();
+
+        let mut x = Tensor::new(&[1]);
+        x[0] = 2i32;
+
+        let x_handle = x.to_handle(&ctx).unwrap();
+
+        let y = unsafe {
+            let status = Status::new();
+            let tf_tensor = tf::TFE_TensorHandleResolve(x_handle.inner, status.inner);
+            Tensor::<i32>::from_tf_tensor(tf_tensor).unwrap()
+        };
+
+        assert_eq!(y[0], 2i32);
+    }
+
+    #[test]
+    fn add_test() {
+        let opts = ContextOptions::new();
+        let ctx = Context::new(opts).unwrap();
+
+        let x = Tensor::from(2);
+        let y = x.clone();
+
+        let h = raw_ops::add(&ctx, x, y).unwrap();
+        let z: Result<Tensor<i32>> = h.resolve();
+        assert!(z.is_ok());
```

Comment:
The `unwrap()` would fail the test anyway if it weren't `ok`, so this assertion isn't really needed.

Reply:
Fixed.